### PR TITLE
[JENKINS-30243] Fix optinality testing of multiple-scms plugin presence

### PR DESCRIPTION
**Pull request by alecharp**
Fixes [JENKINS-30243](https://issues.jenkins-ci.org/browse/JENKINS-30243)

### CODE REVIEW

**Comment by alecharp** [comment]
@reviewbybees

File: src/main/java/hudson/plugins/templateproject/ProxySCM.java [hunk 1]
```diff
@@ -82,7 +78,7 @@ public void checkout(@Nonnull Run<?,?> build, @Nonnull Launcher launcher, @Nonnu
 		// https://github.com/jenkinsci/multiple-scms-plugin/pull/6
 		// https://issues.jenkins-ci.org/browse/JENKINS-27638
 		// Since MultiSCM is optional, might not be installed, hacky check string name.
-		if (getProjectScm((AbstractBuild) build).toString().contains("multiplescms")) {
+		if(Jenkins.getInstance() != null && Jenkins.getInstance().getPlugin("multiple-scms") != null) {
```
Comment (MRamonLeon) [inline src/main/java/hudson/plugins/templateproject/ProxySCM.java:81]: Could you use ```Jenkins.get()```, because ```getInstance()``` is deprecated. Don't need to test for null

File: src/main/java/hudson/plugins/templateproject/ProxySCM.java [hunk 2]
```diff
@@ -106,7 +102,7 @@ public boolean pollChanges(AbstractProject project, Launcher launcher,
 		return getProjectScm().pollChanges(project, launcher, workspace, listener);
 	}
 
-	@Extension
+	@Extension(optional = true)
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/templateproject/ProxySCM.java:105]: What do you expect to happen here?

* SCMDescriptor is in the core, it always exists
* There is no static initializer which would break classloading if a class is missing

File: src/main/java/hudson/plugins/templateproject/ProxySCM.java (hunk 1 above)
Comment (alecharp) [reply src/main/java/hudson/plugins/templateproject/ProxySCM.java:81]: It doesn't exists in the minimum core requirement of this plugin. I would if I could.

File: src/main/java/hudson/plugins/templateproject/ProxySCM.java (hunk 2 above)
Comment (alecharp) [reply src/main/java/hudson/plugins/templateproject/ProxySCM.java:105]: The class is using `org.jenkinsci.plugins.multiplescms.MultiSCMRevisionState` from Multiple SCMs plugin. But `multiple-scms` plugin is an optional dependency of Template project plugin.

With this patch, we now have

```
hudson.ExtensionFinder$GuiceFinder$FaultTolerantScope$1 error
INFO: Failed to instantiate optional component hudson.plugins.templateproject.ProxySCM$DescriptorImpl; skipping
```

with no stack trace.

Would the second part of the patch be enough? I didn't think so, aka I might be wrong.

File: src/main/java/hudson/plugins/templateproject/ProxySCM.java (hunk 1 above)
Comment (fcojfernandez) [reply src/main/java/hudson/plugins/templateproject/ProxySCM.java:81]: not 100% sure, but I think he can not do it. The parent pom is too old, so that method probably is not available for this plugin. @alecharp, have you consider upgrade the parent pom to a newer version?

Comment (alecharp) [reply src/main/java/hudson/plugins/templateproject/ProxySCM.java:81]: I prefer not upgrading the parent pom nor Jenkins Core version requirement if it's not necessary for the PR. Limiting the impact of the changeset.

File: src/main/java/hudson/plugins/templateproject/ProxySCM.java (hunk 2 above)
Comment (oleg-nenashev) [reply src/main/java/hudson/plugins/templateproject/ProxySCM.java:105]: OK, nice that it works as is. I assumed that it may ignore the parent class initialization in this case

File: src/main/java/hudson/plugins/templateproject/ProxySCM.java (hunk 1 above)
Comment (fcojfernandez) [reply src/main/java/hudson/plugins/templateproject/ProxySCM.java:81]: Fair enough

**Review by fcojfernandez — APPROVED** [review]

File: src/main/java/hudson/plugins/templateproject/ProxySCM.java (hunk 2 above)
Comment (alecharp) [reply src/main/java/hudson/plugins/templateproject/ProxySCM.java:105]: I tested with an empty Jenkins. I could have break more than I wanted. So now I'm scared.

**Review by MRamonLeon — APPROVED** [review]

**Comment by alecharp** [comment]
@Brantone are you still maintaining this repository?

**Review by oleg-nenashev — APPROVED** [review]

**Comment by kave** [comment]
Any idea when this is going to be merged?

**Comment by ctran** [comment]
Is this plugin still being maintained or can we get it adopted?

**Comment by alecharp** [comment]
It seems that the project could be adopted. If you are interesting to do that @ctran, could you send an email to the dev mailing list as described in https://www.jenkins.io/doc/developer/plugin-governance/adopt-a-plugin/? Thanks

**Comment by alecharp** [comment]
The plugin seems to be un-maintained. I'm closing this pull request, next maintainer might consider re-opening it or fixing the problem directly.
